### PR TITLE
Modify API for external usage

### PR DESCRIPTION
**Pull request by DeV1doR**
- transaction;
 - node_key;
 - copied methods from v1;

### CODE REVIEW

**Comment by apadenko** [comment]
I have one concern about API methods configuration. 
How will we manage different versions in method switcher ? 
Maybe several changes will be required in enviroment variable which will be parsed.

Did you test cases of enabling/disabling some methods from different version ?

**Comment by DeV1doR** [comment]
@apadenko If functionality will be dropped off from v1 (mean that code remains but api will not be available), it will works.
As I saw, we do not have hardcoded API version in method switcher.
Let's discuss a part with env part.

- Did you test cases of enabling/disabling some methods from different version ?
Tested cases when we remove api from the openapi.yml, it still works.

**Comment by DeV1doR** [comment]
As a proposition, in v2 version use another base structure for payloads in the API so that it is easier to expand it. (proposition was to use like as ws message data structure in the ticket https://remmeio.atlassian.net/browse/REM-61)

File: remme/rest_api/v2/transaction.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import base64
+from contextlib import suppress
+
+from google.protobuf.message import DecodeError
+from sawtooth_sdk.protobuf.transaction_pb2 import Transaction
+
+from remme.account.client import AccountClient
+
+
+def post(payload):
```
[the remark this answers is not in the document]
Comment (eugene-babichenko) [reply remme/rest_api/v2/transaction.py]: I suggest write tests right in this module (like `rest_api.v2.tests`). Before that I will write a proposal on how to test REST APIs.

[the remark this answers is not in the document]
Comment (DeV1doR) [reply remme/rest_api/v2/transaction.py]: @eugene-babichenko will do

**Comment by DeV1doR** [comment]
@apadenko REMME_REST_API_AVAILABLE_METHODS will be work with routes when we add them an api version directly (ex: /v1/certificate:POST;/v2/certificate:*)

File: remme/rest_api/transaction.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+def post(payload):
+    tr = payload['transaction']
+    with suppress(Exception):
+        tr = tr.encode('utf-8')
+
+    try:
+        transaction = base64.b64decode(tr)
+    except Exception:
+        return {'error': 'Decode payload of tranasaction failed'}, 500
```
Comment (eugene-babichenko) [inline remme/rest_api/transaction.py]: Maybe error code 400? Same for transaction proto.

Comment (DeV1doR) [reply remme/rest_api/transaction.py]: Agree

File: remme/atomic_swap_tp/rest_api.py [hunk 3]
```diff
@@ -5,47 +5,51 @@
...
 
-client = AtomicSwapClient()
-
 LOGGER = logging.getLogger(__name__)
 
 
 def init(**data):
     data = data['payload']
     payload = get_swap_init_payload(**data)
+    client = AtomicSwapClient()
```
Comment (eugene-babichenko) [inline remme/atomic_swap_tp/rest_api.py]: Why do we need this here and hereafter?

File: remme/tests/test_rest_api.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import time
```
Comment (eugene-babichenko) [inline remme/tests/test_rest_api.py]: Those dumb transactions will definitely fail. Why don't we test the API against a particular correct transaction and test if it works?

Comment (DeV1doR) [reply remme/tests/test_rest_api.py]: I will add func tests.

File: remme/atomic_swap_tp/rest_api.py (hunk 3 above)
Comment (DeV1doR) [reply remme/atomic_swap_tp/rest_api.py]: It had some problem with a node key when I tested on py.test. Will undo this.

File: remme/tests/test_rest_api.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        return transaction
+
+    @test
+    def test_node_key_retrieve_info_and_it_ok(self):
+        response = self.client.get('/api/v1/node_key')
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('pubkey' in response.get_json())
+
+    @test
+    @mock.patch('remme.shared.basic_client.BasicClient._send_request',
```
Comment (DeV1doR) [inline remme/tests/test_rest_api.py]: @eugene-babichenko uncomment this one and remove 'req_mock' for test raw transaction to the rest api.

File: remme/tests/test_rest_api.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        response = self.client.post('/api/v1/transaction',
+                                    data=json.dumps({
+                                        'transaction': base64.b64encode(payload.SerializeToString()).decode('utf-8')
+                                    }),
+                                    content_type='application/json')
+        self.assertEqual(response.status_code, 200, 'Error: %s' % response.get_data())
+        self.assertTrue('batch_id' in response.get_json())
+
+    @test
+    @mock.patch('remme.shared.basic_client.BasicClient._send_request',
```
Comment (DeV1doR) [inline remme/tests/test_rest_api.py]: @eugene-babichenko comment this one and remove 'req_mock' for test raw transaction to the rest api.

**Pull request merged**